checkbox-support/checkbox_support/snap_utils/snapd.py

```diff
@@ -89,8 +89,27 @@ def _poll_change(self, change_id):
                 abort_result = self._abort_change(change_id)
                 raise AsyncException(status, abort_result)
             for task in self.tasks(change_id):
-                if task['status'] == 'Doing':
-                    self._info(task['summary'])
+                if task["status"] == "Doing":
```

Comment:
I think there are a few issues here, [the snapd documentation](https://snapcraft.io/docs/snapd-api)  (under Task Fields) the status can be: 
- "Do" - Task ready to start.
- "Doing" - Task in progress.
- "Done" - Task is complete.
- "Abort" - Task has been aborted.
- "Undo" - Task needs to be undone.
- "Undoing" - Task is being undone.
- "Hold" - Task will not be run (probably due to failure of another task).
- "Error" - Task completed with an error.

So two main issues: 
- Wait is not there, so I don't know if its undocumented or we shouldn't handle it
- What about the others, I feel like they are all pretty useful to track the status of the task, should we just dump the "summary" in stdout?

Reply:
Also, returning from this function is correct? I would expect that if I call a poll function it will either return on success or raise an exception on failure, while from the commit history you seem to hint to the fact that wait is a non recoverable failure and expects a reboot.

Reply:
Resolved in dm